src/core/modules/Planet.cpp

```diff
@@ -907,12 +1009,66 @@ QString Planet::getInfoStringExtra(const StelCore *core, const InfoStringGroup&
 			{
 				oss << QString("%1: %2%").arg(q_("Eclipse obscuration")).arg(QString::number(eclipseObscuration, 'f', 2)) << "<br />";
 				PlanetP obj = eclObj.second;
-				if (onEarth && obj==ssystem->getMoon())
+				if (onEarth && obj == ssystem->getMoon())
+				{
+					const double eclipseMagnitude =
+					  (0.5 * angularSize
+						+ (obj->getAngularSize(core) * M_PI_180) / obj->getInfoMap(core)["scale"].toDouble()
+						- getJ2000EquatorialPos(core).angle(obj->getJ2000EquatorialPos(core)))
+					  / angularSize;
+					oss << QString("%1: %2")
+							 .arg(q_("Eclipse magnitude"))
+							 .arg(QString::number(eclipseMagnitude, 'f', 3))
+						<< "<br />";
+				}
+			}
+		}
+
+		if (englishName == "Sun" && onEarth)
+		{
+			// Solar eclipse information
+			// Use geocentric coordinates
+			StelCore* core1 = StelApp::getInstance().getCore();
+			const bool useTopocentric = core1->getUseTopocentricCoordinates();
+			core1->setUseTopocentricCoordinates(false);
+			core1->update(0);
+
+			double raSun, deSun, raMoon, deMoon;
+			StelUtils::rectToSphe(&raSun, &deSun, ssystem->getSun()->getEquinoxEquatorialPos(core1));
+			StelUtils::rectToSphe(&raMoon, &deMoon, ssystem->getMoon()->getEquinoxEquatorialPos(core1));
+
+			// R.A. of the Sun
+			raSun = raSun / M_PI_180;
+			if (raSun < 0.) raSun += 360.;
+			// R.A. of the Moon
+			raMoon = (raMoon / M_PI_180);
+			if (raMoon < 0.) raMoon += 360.;
+
+			double raDiff = raMoon - raSun;
+			if (raDiff < 0.) raDiff += 360.;
+
+			if (raDiff < 3. || raDiff > 357.)
+			{
+				SolarEclipse center;
+				Vec3d pos = center.point(core1);
+
+				if (pos[0] < 90.) // only display when shadow axis is touching Earth
 				{
-					const double eclipseMagnitude = (0.5*angularSize + (obj->getAngularSize(core)*M_PI_180)/obj->getInfoMap(core)["scale"].toDouble() - getJ2000EquatorialPos(core).angle(obj->getJ2000EquatorialPos(core)))/angularSize;
-					oss << QString("%1: %2").arg(q_("Eclipse magnitude")).arg(QString::number(eclipseMagnitude, 'f', 3)) << "<br />";
+					oss << QString("%1: %2%3/%4%5")
+							 .arg(q_("Center of solar eclipse (Lat./Long.)"))
+							 .arg(pos[0], 5, 'f', 4)
+							 .arg(QChar(0x00B0))
+							 .arg(pos[1], 5, 'f', 4)
+							 .arg(QChar(0x00B0));
+					oss << "<br/>";
+					oss << QString("%1: %2")
+							 .arg(q_("Magnitude at center line"))
+							 .arg(QString::number(pos[2], 'f', 4))
```

Comment:
It may be useful to add "(total)" or "(annular)" after the number, to make it better understandable.

Reply:
It may be "(hybrid)" also..

Reply:
At any given time, it is either total or annular. "hybrid" is a rather new term (the classic being "annular-total") and describes an eclipse which switches characteristic as the shadow just crosses the limit.